tedana/selection/selection_utils.py

```diff
@@ -1,9 +1,14 @@
 """Utility functions for tedana.selection."""
 
+import copy
 import logging
+import re
+from typing import Dict, List, Tuple, Union
 
 import numpy as np
+import pandas as pd
 
+# from tedana.selection.component_selector import ComponentSelector
```

Comment:
@tsalo This is a point where I'm having trouble with type hints. There are several functions that take the ComponentSelector object as a parameter or return it. I need to import ComponentSelector to have it as an option for a type hint, but I can't because ComponentSelector uses functions in this file and creates a circularity. There seem to be type hint aliases and other mechanisms in python 3.8, but I'm not sure exactly how to use them.

I've also had to import pandas in a few places solely to allow `pd.DataFrame` and `pd.DataSeries` as type hints. I'm not sure that the correct way to do this.

Reply:
Oof that's tough. I guess you can't have a type hint for those functions. Can you just note that in the docstrings?

>I've also had to import pandas in a few places solely to allow pd.DataFrame and pd.DataSeries as type hints. I'm not sure that the correct way to do this.

I think that's fine. Pandas is going to be imported when running tedana anyway, so I don't think it costs us anything to import it in certain files just for type hints.